Helm_charts (#182)
percona/everest-doc: docs/install/install_everest_helm_charts.md

    ```

2. Install Percona Everest:

    ```sh
    helm install everest-core percona/everest \
    --namespace everest-system \
    --create-namespace
    ```

    ??? info "What's happening under the hood"
        The command does the following:
        {.power-number}

        1. Deploys the Percona Everest components in the `everest-system` namespace. Currently, specifying a different namespace for Percona Everest is not supported.

        2. Deploys a new namespace called `everest` for your databases and the database operators.

            You can override the name of the database namespace by using the `dbNamespace.namespaceOverride` parameter. If you prefer to deploy just the core components, set `dbNamespace.enabled=false`

3. Once the installation is complete, retrieve the `admin` password:

    ```sh
    kubectl get secret everest-accounts -n everest-system -o jsonpath='{.data.users\.yaml}' | base64 --decode  | yq '.admin.passwordHash'
    ```

4. Connect to the Percona Everest UI by setting up port-forwarding to your local machine:

    ```sh
    kubectl port-forward svc/everest -n everest-system 8080:8080

>     ```

Reviewer: We should also mention the other methods (NodePort, LoadBalancer) of accessing everest as we do in the everestctl installation page.

Author: Yes. I am still working on it.

> # Install Percona Everest using Helm
> 
> Helm is a package manager for Kubernetes applications.
> 
> The Percona Everest Helm chart can be set up with a single command and includes all the resources required to deploy and manage Percona Everest on your Kubernetes cluster. 
> 
> The Helm chart simplifies the Percona Everest deployment process by packaging the configuration, dependencies, and templates. This streamlined approach ensures that everything can be set up smoothly without any hassle. For more information on Helm, refer to the official [documentation](https://helm.sh/docs/topics/charts/){:target="_blank"}.
> 
> ## Install Percona Everest
> 
> You can install Percona Everest using Helm as an alternative method.
> {.power-number}

Reviewer 2: I know we already spoke about this a few weeks back, but I still think a lot of this information is redundant, or not required. It makes it sound like the listed benefits are unique to the Everest Helm chart, but in reality is just a standard thing offered by Helm 😅 

I would go with something more concise..

 
```suggestion
# Install Percona Everest using Helm
This guide demonstrates how to install Percona Everest using [Helm](https://helm.sh/) as an alternative to `everestctl install`. Helm charts simplify deployment by packaging all required resources and configurations, making it an ideal choice for automating and managing installations in Kubernetes environments.

For more info, refer to the official documentation on [Helm charts](https://helm.sh/docs/topics/charts/#helm).

## Install Percona Everest
```

As reference, I'll attach how other cloud-native projects document their Helm install guides:
- https://linkerd.io/2-edge/tasks/install-helm/
- https://docs.cilium.io/en/stable/installation/k8s-install-helm/
- https://docs.gitlab.com/charts/installation/
- https://falco.org/docs/setup/kubernetes/

.. as you can see, the info and benefits about helm itself is deferred to the official docs, and these only focus on how to use the chart

    --namespace everest-system \
    --create-namespace
    ```

    ??? info "What's happening under the hood"
        The command does the following:
        {.power-number}

        1. Deploys the Percona Everest components in the `everest-system` namespace. Currently, specifying a different namespace for Percona Everest is not supported.

        2. Deploys a new namespace called `everest` for your databases and the database operators.

            You can override the name of the database namespace by using the `dbNamespace.namespaceOverride` parameter.

3. Once the installation is complete, retrieve the `admin` password:

    ```sh
    kubectl get secret everest-accounts -n everest-system -o jsonpath='{.data.users\.yaml}' | base64 --decode  | yq '.admin.passwordHash'
    ```

4. Connect to the Percona Everest UI by setting up port-forwarding to your local machine:

    ```sh
    kubectl port-forward svc/everest -n everest-system 8080:8080
    ```

    !!! note

        - The default username for logging into the Everest UI is `admin`.
        - You can set a different default admin password by using the `server.initialAdminPassword` parameter during installation.

>         - The default admin password is stored in plain text. It is highly recommended to update the password using `everestctl` to ensure that the passwords are hashed.

Author: I think it makes sense to keep part of the install section only.

A few parameters are listed in the following table. For a detailed list of the parameters, see the [README](https://github.com/percona/percona-helm-charts/blob/main/charts/everest/README.md#configuration){:target="_blank"}.

|**Namespace**|**Key**|**Type**|**Default**|**Description**|
|---------|------|---------|------------------|------------|
|everest|`server.initialAdminPassword`|string|""|Initial password configured for admin user.</br></br> If it is not set, a random password is generated. It is recommended to reset the admin password after installation.|
|everest|`server.oidc`|object|{}|OIDC configuration for Everest.</br></br> These settings are applied only during installation. To modify the settings after installation, you have to manually update the everest-settings `ConfigMap`.|
|everest-db-namespace|`pxc`|bool|true| Installs the Percona XtraDB Cluster operator if set.|
|everest-db-namespace|`postgresql`|bool|true| Installs the Percona Postgresql Server operator if set.|

> |everest-db-namespace|`psmdb`|bool|true| Installs the Percona Server MongoDB operator if set.|

Author: Fixed

            v1.30.3-gke.1969001   10.204.15.201   34.175.201.246   Container- 
            Optimized OS from Google   6.1.100+         containerd://1.7.19
            ```
        
        4. To launch the Percona Everest UI and create your first database cluster, go to the IP address/port found in steps 2 and 3. In this example, the external IP address used is [http://34.175.155.135:32349](http://34.175.155.135:32349). Nevertheless, you have the option to use any node IP specified in the above steps.

    === "Port Forwarding"
        Run the following command to use `Kubectl port-forwarding` for connecting to Everest without exposing the service:
                
        ```sh
        kubectl port-forward svc/everest 8080:8080 -n everest-system
        ``` 

        To launch the Percona Everest UI and create your first database cluster, go to your localhost IP address [http://127.0.0.1:8080](http://127.0.0.1:8080).

5. The default username for logging into the Everest UI is `admin`. You can set a different default admin password by using the `server.initialAdminPassword` parameter during installation.

>     The default admin password is stored in plain text. It is highly recommended to update the password using `everestctl` to ensure that the passwords are hashed.

Reviewer: We need a link to the user management section.

Author: Fixed.

5. Deploy additional database namespaces:

    Once Percona Everest is successfully running, you can create additional database namespaces using the `everest-db-namespace` Helm chart. 

>     If you set `dbNamespaces.enabled=false` in the previous step, you can deploy a database namespace with the following command:

Reviewer: ```suggestion
    If you set `dbNamespaces.enabled=false` in step 2., you can deploy a database namespace with the following command:
```

    ```sh
    helm install everest \
    percona/everest-db-namespace \
    --create-namespace \

>     --namespace everest

Reviewer: Let's keep this open for users to decide:
```suggestion
    --namespace <DBNamespace>
```

    
    For more details, refer to the [Container Registry Deprecation documentation](https://cloud.google.com/artifact-registry/docs/transition/prepare-gcr-shutdown){:target="_blank"}.

This section explains how to install Percona Everest using [Helm](https://helm.sh/) as an alternative to `everestctl`. Helm charts simplify the deployment process by packaging all necessary resources and configurations, making them ideal for automating and managing installations in Kubernetes environments.

Percona Helm charts can be found in [percona/percona-helm-charts]( https://github.com/percona/percona-helm-charts/tree/main/charts/everest) repository in Github.

## Install Percona Everest and deploy database namespaces

Here are the steps to install Percona Everest and deploy additional database namespaces:
{.power-number}

1. Add the Percona Helm repository:

    ```sh

>     helm repo add percona https://percona.github.io/percona-helm-charts/

Reviewer 3: Is missing `helm repo update`

Author: Fixed.

            ```
        
        4. To launch the Percona Everest UI and create your first database cluster, go to the IP address/port found in steps 2 and 3. In this example, the external IP address used is [http://34.175.155.135:32349](http://34.175.155.135:32349). Nevertheless, you have the option to use any node IP specified in the above steps.

    === "Port Forwarding"
        Run the following command to use `Kubectl port-forwarding` for connecting to Everest without exposing the service:
                
        ```sh
        kubectl port-forward svc/everest 8080:8080 -n everest-system
        ``` 

        To launch the Percona Everest UI and create your first database cluster, go to your localhost IP address [http://127.0.0.1:8080](http://127.0.0.1:8080).

5. The default username for logging into the Everest UI is `admin`. You can set a different default admin password by using the `server.initialAdminPassword` parameter during installation.

    The default `admin` password is stored in plain text. It is highly recommended to update the password using `everestctl` to ensure that the passwords are hashed.

>     To access detailed information on user management, see the [manage users in Percona Everest](../administer/manage_users.md#update-the-password) section.

Reviewer: Maybe this should be merged with step 3. where we tell users to retrieve the admin password, similarly with what we do in the CLI installation.

Author: Done!

    If you set `dbNamespaces.enabled=false` in the previous step, you can deploy a database namespace with the following command:

    ```sh
    helm install everest \
    percona/everest-db-namespace \
    --create-namespace \
    --namespace everest
    ```

    !!! note
        -  All database operators are installed in your database namespace by default. You can override this by specifying one or more of the following options: `[dbNamespace.pxc=false, dbNamespace.pg=false, dbNamespace.psmdb=false]`.
        - Installation without chart hooks (i.e, the use of `--no-hooks`) is currently not supported.

## Configure parameters

You can customize various parameters in the Percona Everest Helm charts for your deployment to meet your specific needs. Refer to the [Helm documentation](https://helm.sh/docs/chart_best_practices/values/){:target="_blank"} to discover how to configure these parameters.

A few parameters are listed in the following table. For a detailed list of the parameters, see the [README](https://github.com/percona/percona-helm-charts/blob/main/charts/everest/README.md#configuration){:target="_blank"}.

> |**Charts**|**Key**|**Type**|**Default**|**Description**|
> |---------|------|---------|------------------|------------|
> |everest|`server.initialAdminPassword`|string|""|Initial password configured for admin user.</br></br> If it is not set, a random password is generated. It is recommended to reset the admin password after installation.|
> |everest|`server.oidc`|object|{}|OIDC configuration for Everest.</br></br> These settings are applied only during installation. To modify the settings after installation, you have to manually update the everest-settings `ConfigMap`.|
> 
> |**Subcharts**|**Key**|**Type**|**Default**|**Description**|
> |---------|------|---------|------------------|------------|
> |everest-db-namespace|`pxc`|bool|true| Installs the Percona XtraDB Cluster operator if set.|
> |everest-db-namespace|`postgresql`|bool|true| Installs the Percona Postgresql Server operator if set.|
> |everest-db-namespace|`psmdb`|bool|true| Installs the Percona Server MongoDB operator if set.|

Reviewer 2: If you're creating 2 separate tables, the `Charts` and `Subcharts` columns are not needed

```suggestion
For `percona/everest` chart:
|**Key**|**Type**|**Default**|**Description**|
|------|---------|------------------|------------|
|`server.initialAdminPassword`|string|""|Initial password configured for admin user.</br></br> If it is not set, a random password is generated. It is recommended to reset the admin password after installation.|
|`server.oidc`|object|{}|OIDC configuration for Everest.</br></br> These settings are applied only during installation. To modify the settings after installation, you have to manually update the everest-settings `ConfigMap`.|
 
For `percona/everest-db-namespace` subchart
|**Key**|**Type**|**Default**|**Description**|
|------|---------|------------------|------------|
|`pxc`|bool|true| Installs the Percona XtraDB Cluster operator if set.|
|`postgresql`|bool|true| Installs the Percona Postgresql Server operator if set.|
|`psmdb`|bool|true| Installs the Percona Server MongoDB operator if set.|
```

# Install Percona Everest using Helm

This section explains how to install Percona Everest using [Helm](https://helm.sh/) as an alternative to `everestctl`. Helm charts simplify the deployment process by packaging all necessary resources and configurations, making them ideal for automating and managing installations in Kubernetes environments.

> Percona Helm charts can be found in [percona/percona-helm-charts](https://github.com/percona/percona-helm-charts) repository in Github.

Reviewer 2: Shouldn't we link to the Percona Everest Helm chart instead?

Author: ok

Author: Fixed

> ## Install Percona Everest and deploy namespaces

Reviewer 2: ```suggestion
## Install Percona Everest and deploy database namespaces
```

> Here are the steps to install Percona Everest and deploy additional namespaces:

Reviewer 2: ```suggestion
Here are the steps to install Percona Everest and deploy additional database namespaces:
```

    If you set `dbNamespaces.enabled=false` in the previous step, you can deploy a database namespace with the following command:

    ```sh
    helm install everest \
    percona/everest-db-namespace \
    --create-namespace \
    --namespace everest
    ```

    !!! note
        -  All database operators are installed in your database namespace by default. You can override this by specifying one or more of the following options: `[dbNamespace.pxc=false, dbNamespace.pg=false, dbNamespace.psmdb=false]`.
        - Installation without chart hooks (i.e, the use of `--no-hooks`) is currently not supported.

## Configure parameters

You can customize various parameters in the Percona Everest Helm charts for your deployment to meet your specific needs. Refer to the [Helm documentation](https://helm.sh/docs/chart_best_practices/values/){:target="_blank"} to discover how to configure these parameters.

A few parameters are listed in the following table. For a detailed list of the parameters, see the [README](https://github.com/percona/percona-helm-charts/blob/main/charts/everest/README.md#configuration){:target="_blank"}.

> **Percona Everest chart**

Reviewer 2: To be consistent with the heading below..

```suggestion
**percona/everest chart**
```

|**Key**|**Type**|**Default**|**Description**|
|------|---------|-----------|---------------|
|`server.initialAdminPassword`|string|""|Initial password configured for admin user.</br></br> If it is not set, a random password is generated. It is recommended to reset the admin password after installation.|
|`server.oidc`|object|{}|OIDC configuration for Everest.</br></br> These settings are applied only during installation. To modify the settings after installation, you have to manually update the everest-settings `ConfigMap`.|

**percona/everest-db-namespace subchart**

|**Key**|**Type**|**Default**|**Description**|
|-------|--------|-----------|---------------
|`pxc`|bool|true| Installs the Percona XtraDB Cluster operator if set.|
|everest-db-namespace|`postgresql`|bool|true| Installs the Percona Postgresql Server operator if set.|
|`psmdb`|bool|true| Installs the Percona Server MongoDB operator if set.|

> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

Reviewer 2: looks like the unnecessary newlines are added by mistake?
```suggestion
```
